src/software/ai/hl/stp/play/defense/defense_play_fsm.cpp

```diff
@@ -162,3 +210,48 @@ void DefensePlayFSM::setTactics(const Update& event)
 
     event.common.set_tactics(tactics_to_return);
 }
+
+bool DefensePlayFSM::validateCreaseDefenderPosition(const Field& field,
+                                                    const Point& threat_origin)
+{
+    // Test all three alignments by calling findBlockThreatPoint
+    std::vector<TbotsProto::CreaseDefenderAlignment> alignments = {
+        TbotsProto::CreaseDefenderAlignment::CENTRE,
+        TbotsProto::CreaseDefenderAlignment::LEFT,
+        TbotsProto::CreaseDefenderAlignment::RIGHT};
+
+    for (const auto& alignment : alignments)
+    {
+        auto position =
+            CreaseDefenderFSM::findBlockThreatPoint(field, threat_origin, alignment, 1.0);
+
+        if (position.has_value())
+        {
+            // Check if position is on or behind goal line
+            Segment goal_line = Segment(field.friendlyGoal().posXPosYCorner(),
```

Comment:
not sure if this is the correct line, print to make sure. i assume this is meant to be the top of the friendly goal?